### ISSUE TITLE
FORCE_COLOR

### ISSUE THREAD

**Issue by perillo** [jhasse/clicolors#15]
Looking at https://docs.python.org/3.13/using/cmdline.html#controlling-color, I found that Python will uses `FORCE_COLOR` (probably invented by [colors.js](https://www.npmjs.com/package/colors) and improved by [chalk](https://www.npmjs.com/package/chalk).

Searching the internet, there are other projects mentioning `FORCE_COLOR`.

Having two different standards will make implementing color support more complex. Is there a path for an unification?

Thanks.

**Comment by jhasse** [comment]
I'm all for unification. I've already talked with chalk, they weren't interested in changing their variable name (which I can understand): https://github.com/chalk/supports-color/issues/32

I could change the site to recommend `FORCE_COLOR` instead of `CLICOLOR_FORCE`, but what about the programs that already use the latter?

**Comment by perillo** [comment]
> I'm all for unification. I've already talked with chalk, they weren't interested in changing their variable name (which I can understand): [chalk/supports-color#32](https://github.com/chalk/supports-color/issues/32)
> 
> I could change the site to recommend `FORCE_COLOR` instead of `CLICOLOR_FORCE`, but what about the programs that already use the latter?

At this point the only solution is to officially recommend one of them, but suggest to also add support the latter.  If the semantics are the same, then an implementator could easily support both.

By the way: [Zig](https://github.com/ziglang/zig) unfortunately started to use `YES_COLOR` (before it was called ZIG_DEBUG_COLOR`).

**Comment by adamhotep** [comment]
See also [`PYTHON_COLORS`](https://docs.python.org/3.13/using/cmdline.html#envvar-PYTHON_COLORS). The big problem here is that there are so many competing conventions because this has never been standardized. Were a [new standard](https://xkcd.com/927/) to be proposed, something like #19 would likely make the most sense, but even if you have the ears of the BSD, GNU, and OpenGroup (Posix) folks, that's still quite a tall order.

I've been a fan of `CLICOLOR` and `CLICOLOR_FORCE` for a while, as they are (unwritten?) BSD standards. `NO_COLOR` seems like a good third item and at least has a good presentation over at [no-color.org](https://no-color.org/). However, their battle must be fought on a per-program basis and that's going to take _forever_—at least `CLICOLOR` and `CLICOLOR_FORCE` are decently adopted (see #6).

**Comment by Ravencentric** [comment]
Also see: https://force-color.org/

**Comment by baodrate** [comment]
the corresponding issue: donatj/force-color.org#42

**Comment by hugovk** [comment]
> See also [`PYTHON_COLORS`](https://docs.python.org/3.13/using/cmdline.html#envvar-PYTHON_COLORS). The big problem here is that there are so many competing conventions because this has never been standardized.

You don't need to worry about `PYTHON_COLORS`. It is intended as a [fine-grained control](https://docs.python.org/3.13/using/cmdline.html#using-on-controlling-color) for the Python interpreter itself:

> All these environment variables are used also by other tools to control color output. To control the color output only in the Python interpreter, the [`PYTHON_COLORS`](https://docs.python.org/3.13/using/cmdline.html#envvar-PYTHON_COLORS) environment variable can be used. This variable takes precedence over `NO_COLOR`, which in turn takes precedence over `FORCE_COLOR`.

There are other `PYTHON_*` env vars intended for the interpreter only. This is similar to pytest which uses `PY_*` env vars, and also has [`PY_COLORS`](https://docs.pytest.org/en/stable/reference/reference.html#envvar-PY_COLORS) for more fine-grained control over `NO_COLOR` and `FORCE_COLOR`.

So you wouldn't need to document app-specific things like `PYTHON_COLORS` or `PY_COLORS` on this site.

(I was involved in adding these to Python and pytest.)

**Comment by jhasse** [comment]
I'm thinking about just redirecting to force-color.org ...
Although CLICOLOR_FORCE was first, it's way more important that there is only one standard.

Fighting for this cause (it's just a stupid env var how hard can it be ...) hasn't been fun anyway, so I'm happy if other people do it.

**Comment by baodrate** [comment]
I think it's useful to have around if only as documentation for current support. I agree that it looks like `FORCE_COLOR` is the path forward for new applications, tho.

Perhaps a banner at the top directing readers to force-color.org as a recommendation?

**Comment by jhasse** [comment]
Yes sounds good. PR welcome if someone wants to do it, otherwise I will try to find some time in the future.

**Comment by hugovk** [comment]
Please see PR https://github.com/jhasse/clicolors/pull/28.
